src/commands/setup.ts

```diff
@@ -0,0 +1,55 @@
+import { Firestore, WriteBatch } from '@google-cloud/firestore';
+import { Client, Message } from 'discord.js';
+
+exports.run = async (client: Client, msg: Message, args: string[], firestore: Firestore) => {
+    if (msg.member!.permissions.has('ADMINISTRATOR') ) { // Ensures only admins may use this command
```

Comment:
Personal preference: I think early exists look cleaner when writing longer form functions like this. So in this case, you would check the inverse instead and if true, then send the error msg and return immeaditely. It help with readability overall, especially in this case.